packages/server/src/workers/cron.ts

```diff
@@ -225,13 +225,11 @@ export async function execBot(job: Job<CronJobData>): Promise<void> {
   await createAuditEvent(bot, startTime, outcome, logResult);
 }
 
-async function removeBullMQJobByKey(botId: string): Promise<void> {
-  const previousJobs = await queue?.getRepeatableJobs();
-  previousJobs?.map(async (p) => {
-    if (p.key.includes(botId)) {
-      const keyToRemove = p.key;
-      await queue?.removeRepeatableByKey(keyToRemove);
-      logger.debug(`Found a previous job for bot ${botId}, updating...`);
-    }
-  });
+export async function removeBullMQJobByKey(botId: string): Promise<void> {
+  const previousJobs = (await queue?.getRepeatableJobs())?.filter((p) => p.id === botId) ?? [];
```

Comment:
Sample  output of `console.log(await queue?.getRepeatableJobs());`

```
[
  {
    key: 'CronJobData:9f13c788-5b32-4d86-a878-758f2f764e70:::0 */12 * * *',
    name: 'CronJobData',
    id: '9f13c788-5b32-4d86-a878-758f2f764e70',
    endDate: null,
    tz: null,
    pattern: '0 */12 * * *',
    next: 1681282800000
  }
]
```

Reply:
Oh nice! This is way better